spec/models/service_interaction_spec.rb

```diff
@@ -1,12 +1,33 @@
 require 'rails_helper'
 
 RSpec.describe ServiceInteraction, type: :model do
-  it { is_expected.to validate_presence_of(:service_identifier) }
-  it { is_expected.to validate_presence_of(:service_label) }
-  it { is_expected.to validate_uniqueness_of(:service_identifier) }
-  it { is_expected.to validate_uniqueness_of(:service_label) }
-  it { is_expected.to validate_presence_of(:interaction_identifier) }
-  it { is_expected.to validate_presence_of(:interaction_label) }
-  it { is_expected.to validate_uniqueness_of(:interaction_identifier) }
-  it { is_expected.to validate_uniqueness_of(:interaction_label) }
+  describe 'validations' do
+    it { is_expected.to validate_presence_of(:service_identifier) }
+    it { is_expected.to validate_presence_of(:service_label) }
+    it { is_expected.to validate_presence_of(:interaction_identifier) }
+    it { is_expected.to validate_presence_of(:interaction_label) }
+
+    it { is_expected.to validate_uniqueness_of(:service_identifier) }
+    it { is_expected.to validate_uniqueness_of(:service_label) }
+    it { is_expected.to validate_uniqueness_of(:interaction_identifier) }
+    it { is_expected.to validate_uniqueness_of(:interaction_label) }
+  end
+
+  describe '#find_service' do
+    it 'returns the correct service from the Services database' do
+      service = FactoryGirl.create(:service, label: 'Abandoned shopping trolleys', identifier: 1152)
+      service_interaction = FactoryGirl.create(:service_interaction)
+
+      expect(ServiceInteraction.find_service(service_interaction.service_identifier)).to eq(service)
```

Comment:
It would be nice to be more explicit in the specs to make it clear *why* we expect the method to return that particular service instance.  As it stands we could make this spec pass if the implementation was just to return `Service.first`.  For example we could be more explicit when building the `service` and `service_integration` instances so that they share identifiers, we could also add a negative test to show that it won't just return any `Service` instance, but only the one that matches on the identifier.

In general it's worth being explicit in our specs about what we're testing, that goes for the setup of the objects we use, but also for the name of the spec too.  Here we're just saying "returns the correct service" but we haven't clearly stated what "correct" means.  It makes it easier to come back to things later and work out what we meant - I'll often look at the specs for an object to discover what it's supposed to do before I look at the object implementation itself as specs can be clearer to read because they tend to be simpler (set up some objects, do a thing with one or more of them, check the results) compared to the actual implementations which can be super complex.

This comment goes for the `find_interaction` spec too.